hbase-oss/src/main/java/org/apache/hadoop/hbase/oss/Constants.java

```diff
@@ -18,9 +18,11 @@
 
 package org.apache.hadoop.hbase.oss;
 
-import org.apache.hadoop.classification.InterfaceAudience;
-import org.apache.hadoop.classification.InterfaceStability;
+import org.apache.yetus.audience.InterfaceAudience;
+import org.apache.yetus.audience.InterfaceStability;
 
+@InterfaceAudience.LimitedPrivate({"HBase"})
```

Comment:
LP (HBase) is meant to reflect used by HBase programatically or in configs? or both?

Reply:
The design allows HBase to call "new HBaseObjectStoreSemantics("s3a://bucket/", conf)" and eliminate the need for any of the *.impl configs. But to avoid code changes I've been testing by using configs to inject HBOSS into that layer. So both - but I'm not opposed to just requiring one approach or the other to make things simpler if the HBase community has a strong preference for one.

I'm a little torn on how the TreeLockManager implementations should be annotated - they're intended to be referenced directly by users in configs, but people should be calling the functions directly.

Reply:
HBase defines a LP(CONFIG) for that use.